### PR TITLE
add example multi-model script

### PR DESCRIPTION
**Pull request by mikeyoungPAI**

### CODE REVIEW

**Comment by mikeyoungPAI** [comment]
@a-guiducci added the example script here. Would appreciate your thoughts! :)

**Comment by mikeyoungPAI** [comment]
@a-guiducci updated the multi-model prompt and added a direct file support example. Would appreciate a quick scan :)

**Comment by a-guiducci** [comment]
@mikeyoungPAI looking now! quick one out the gate, the file example file has a typo :)

File: python/LLM Examples/secure_prompt_from_file.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+import json
+import base64
+import os
+
+# Initialize parser
+parser = argparse.ArgumentParser(description="Secure LLM prompting from a file")
+parser.add_argument("-d", "--directory", required=True, help="directory")
+parser.add_argument("-f", "--file", required=True, help="File to redact")
+parser.add_argument("-t", "--filetype", required=True, help="file type")
+parser.add_argument("-n", "--filename", required=True, help="file type")
```
Comment (a-guiducci) [inline python/LLM Examples/secure_prompt_from_file.py:16]:
```suggestion
parser.add_argument("-n", "--filename", required=True, help="file name")
```

File: python/LLM Examples/secure_prompt_from_file.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+import argparse
+import json
+import base64
+import os
+
+# Initialize parser
+parser = argparse.ArgumentParser(description="Secure LLM prompting from a file")
+parser.add_argument("-d", "--directory", required=True, help="directory")
+parser.add_argument("-f", "--file", required=True, help="File to redact")
+parser.add_argument("-t", "--filetype", required=True, help="file type")
```
Comment (a-guiducci) [inline python/LLM Examples/secure_prompt_from_file.py:15]: I would actually only accept the extension itself here (eg. pdf). I'll put a sample below on how to validate it for the deid service so we can make it a tad more robust and won't have to split anything. 
The help could also be updated to show a sample like pdf or whatever just for clarity

File: python/LLM Examples/secure_prompt_from_file.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+def prompt_chat_gpt(text):
+    completion = openai.ChatCompletion.create(
+        model="gpt-4",
+        messages=[
+        {"role": "user",
+         "content": text}
+        ]
+    )
+    return completion.choices[0].message['content']
+
```
Comment (a-guiducci) [inline python/LLM Examples/secure_prompt_from_file.py:43]:
```suggestion
content_type = None
for row in request_objects.file_obj.valid_content_types:
    if file_type == row.slit('/')[-1]:
        content_type = row
if not content_type:
    raise ValueError(f"{file_type} is not a valid file type for deidentification")

```

File: python/LLM Examples/secure_prompt_from_file.py [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+# Read from file
+with open(filepath, "rb") as b64_file:
+    file_data = base64.b64encode(b64_file.read())
+    file_data = file_data.decode("ascii")
+
+# Make the request
+file_obj = request_objects.file_obj(data=file_data, content_type=args.filetype)
```
Comment (a-guiducci) [inline python/LLM Examples/secure_prompt_from_file.py:50]: change the content_type to be content_type if you wanna use the suggestion above

**Review by a-guiducci — APPROVED** [review]
I think it looks good! Super helpful for getting started with llms

I have a couple thoughts, none of them necessary as this is just a sample script :) 
1) I'd apply some oop to this and put each model's logic in a class (all derived from a base class). that way the main function is cleaner as it doesn't have to determine how to structure the prompt or anything, it can call client.send_prompt() or whatever and the class would know how to handle that. 

2) it can be more readable if you declare your starting point with a 
if __name__ == "__main__":
    main()
and then wrap all the global declarations in contextually relevant functions  and call them as needed. The argument parsing can be placed before the main() function call. 

3) It might be beneficial to include a sample config file (or env file or whatever) just so the user can easily determine the necessary variables needed for each client

**Pull request merged**